### PR TITLE
add 2d inputs and copy transpose to transpose benchmark

### PR DESCRIPTION
**Pull request by liqiangxl**
Two extensions to transpose benchmark in `benchmarks/python/test_transpose.py`

**(1) Adds coverage for copy vs. view transpose**
Previously, we only exercised view transpose, which returns a non-contiguous tensor and the pointwise scheduler is used. As a result, the transpose scheduler was never actually used.
This PR adds `.contiguous()` to enforce a contiguous output layout, which triggers a copy-based transpose.
For manually defined fusions, a `segment_set` was added to the fusion to avoid the pre-segmentation pass (`AllocationDomainPass`) changing transpose output layout to ensure the copy transpose path is taken.

**For view transpose, the output has a allocation domain of `(iS11{i0}, iS10{i1})` which is same as input**
```
T5_g_float[iS10{i1}, iS11{i0}]
 logical domain : (iS10{i1}, iS11{i0})
 allocation domain : (iS11{i0}, iS10{i1})
 contiguity: t t
 loop domain : (iS10{i1}, iS11{i0})
```
Final fusion is:
```
Segmented_Fusion{ 
groups: 
  pointwise{0, 1, 2, 3}
edges: 

group details:
g{(pointwise)
group id: 0
inputs:
  T0_g_float[iS0{i0}, iS1{i1}] float
  T1_g_float[iS12{i0}, iS13{i1}] float
outputs:
  T5_g_float[iS10{i1}, iS11{i0}] float


T2_l_float[iS4{i0}, iS5{i1}]
   = T0_g_float[iS0{i0}, iS1{i1}]
   + T1_g_float[iS12{i0}, iS13{i1}];
(0)
T3_l_float[iS7{i1}, iS6{i0}]
   = Set.Permute( T2_l_float[iS4{i0}, iS5{i1}], cache_op=Streaming )
(1)
T4_l_bool[iS8{i1}, iS9{i0}]
   = T3_l_float[iS7{i1}, iS6{i0}]
   > double(0);
(2)
T5_g_float[iS10{i1}, iS11{i0}]
   = where(T4_l_bool[iS8{i1}, iS9{i0}]
  , T3_l_float[iS7{i1}, iS6{i0}]
  , double(0));
(3)
}

} //Segmented_Fusion
```
**For copy transpose, the output is T6,  it has `a transposed allocation domain : (iS12{i1}, iS13{i0})`:**
```
T6_g_float[iS12{i1}, iS13{i0}]
   = SegmenterSet( T5_l_float[iS10{i1}, iS11{i0}] )

T5_l_float[iS10{i1}, iS11{i0}]
 logical domain : (iS10{i1}, iS11{i0})
 contiguity: t t
 loop domain : (iS10{i1}, iS11{i0})
T6_g_float[iS12{i1}, iS13{i0}]
 logical domain : (iS12{i1}, iS13{i0})
 allocation domain : (iS12{i1}, iS13{i0})
 contiguity: t t
 loop domain : (iS12{i1}, iS13{i0})
```

Final fusion is:
```
Segmented_Fusion{ 
groups: 
  transpose{0, 1, 2, 3, 4}
edges: 

group details:
g{(transpose)
group id: 0
inputs:
  T0_g_float[iS0{i0}, iS1{i1}] float
  T1_g_float[iS14{i0}, iS15{i1}] float
outputs:
  T6_g_float[iS12{i1}, iS13{i0}] float


T2_l_float[iS4{i0}, iS5{i1}]
   = T0_g_float[iS0{i0}, iS1{i1}]
   + T1_g_float[iS14{i0}, iS15{i1}];
(0)
T3_g_float[iS7{i1}, iS6{i0}]
   = Set.Permute( T2_l_float[iS4{i0}, iS5{i1}], cache_op=Streaming )
(1)
T4_g_bool[iS8{i1}, iS9{i0}]
   = T3_g_float[iS7{i1}, iS6{i0}]
   > double(0);
(2)
T5_g_float[iS10{i1}, iS11{i0}]
   = where(T4_g_bool[iS8{i1}, iS9{i0}]
  , T3_g_float[iS7{i1}, iS6{i0}]
  , double(0));
(3)
T6_g_float[iS12{i1}, iS13{i0}]
   = SegmenterSet( T5_g_float[iS10{i1}, iS11{i0}] )
(4)
}

} //Segmented_Fusion

```

**(2) Generalizes fusion input ranks to 2D**
Previously, fusion inputs were limited to 3D shapes, with roughly 100 test cases per data type. This PR expands coverage to include 2D input shapes as well.

### CODE REVIEW

**Comment by liqiangxl** [comment]
!test

File: benchmarks/python/test_transpose.py [hunk 1]
```diff
@@ -34,25 +34,55 @@ def transpose_fusion(
...
+        for size in sizes:
+            for axes in axes_list:
+                params.append((size, axes, dims))
+    return params
+
+
+@pytest.mark.parametrize("size,axes,dims", _generate_transpose_params())
 @pytest.mark.parametrize("dtype", FLOAT_DTYPES)
-@pytest.mark.parametrize("axes", [(0, 1), (0, 2), (1, 2)])
+@pytest.mark.parametrize(
```
Comment (Priya2698) [inline benchmarks/python/test_transpose.py:73]: Do we need to benchmark `view` transpose? Should we remove it instead?

File: benchmarks/python/test_transpose.py [hunk 2]
```diff
@@ -65,26 +95,39 @@ def test_transpose_nvf_benchmark(
...
+        )
         fd.validate([input1, input2], [eager_output])
 
     if not disable_benchmarking:
         run_benchmark(benchmark, fd.execute, [input1, input2])
 
 
 @pytest.mark.parametrize("executor", DEFAULT_EXECUTORS)
-@pytest.mark.parametrize("size", generate_input_sizes(dims=3))
+@pytest.mark.parametrize("size,axes,dims", _generate_transpose_params())
```
Comment (Priya2698) [inline benchmarks/python/test_transpose.py:117]: IIRC, I used 3D inputs to match C++ benchmark. If 2D inputs are sufficient for benchmarking, we should remove the 3D benchmarking. This should also simplify the dashboard for this benchmark

File: benchmarks/python/test_transpose.py [hunk 3]
```diff
@@ -34,25 +34,55 @@ def transpose_fusion(
...
     T7 = fd.ops.gt(T5, S6)
     T9 = fd.ops.where(T7, T5, S6)
-
-    fd.add_output(T9)
-
-
-def transpose_fwd_fn(inputs: list):  # [input1, input2, dim0, dim1]
-    return torch.nn.functional.relu(
+    # add segmenter set to avoid presegment passes setting the output as a view of the input without any data movement. It leads to pointwise instead of transpose scheduler.
+    # we can also expose OptimizationPassGuard to python frontend and disable presegment passes to enforce output to be contiguous and then transpose scheduler will be used.
```
Comment (Priya2698) [inline benchmarks/python/test_transpose.py]:
```Suggestion
    #we can also expose OptimizationPassGuard to python frontend and disable presegmentation passes to enforce output to be contiguous and then transpose scheduler will be used.

File: benchmarks/python/test_transpose.py (hunk 1 above)
Comment (liqiangxl) [reply benchmarks/python/test_transpose.py:73]: I don't know, it's not an expensive benchmark, so I just leave it as is in this PR.

File: benchmarks/python/test_transpose.py (hunk 2 above)
Comment (liqiangxl) [reply benchmarks/python/test_transpose.py:117]: should keep 3D for different axes

File: benchmarks/python/test_transpose.py (hunk 1 above)
Comment (Priya2698) [reply benchmarks/python/test_transpose.py:73]: Got it. Please work with @xwang233 for dashboard integration.

**Review by Priya2698 — APPROVED** [review]
LGTM. If you find that the `view` transpose variant is not meaningful anymore, please remove it in a future follow-up.

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
> LGTM. If you find that the `view` transpose variant is not meaningful anymore, please remove it in a future follow-up.

It provides apple-to-apple compare and ensure nvFuser is smart enough to detect and use view transpose.
